### ISSUE TITLE
Include time taken during minimization

### ISSUE THREAD

**Issue by anton164** [stephencookdev/speed-measure-webpack-plugin#49]
In Webpack 4, we specify the minimizer like so:
```
  optimization: {
    minimizer: [
      new TerserPlugin({
        sourceMap: true,
        parallel: true,
        cache: './.build_cache/terser',
        exclude: /transpiledLibs/,
        terserOptions: {
          warnings: false,
          ie8: false
        }
      })
    ]
  },
```

Meaning that it is no longer included in the SMP output time. This would be a nice addition to the plugin :)

**Comment by stephencookdev** [comment]
Ah, yes. This wouldn't be too big a change, it should just be specifying an extra source for proxying.

I'll try and raise a PR this week for the change, unless anyone else wanted to give this one a go :)

**Comment by stephencookdev** [comment]
Thanks @anton164 - would be awesome if you had any time to have a look :smile: 

If you look at https://github.com/stephencookdev/speed-measure-webpack-plugin/blob/master/index.js#L41 you can see how we map through `config.plugins`, and wrap plugin with `WrappedPlugin` 

We should just need to add an extra bit doing similarly for any extra plugins that `webpack.optimization` allows the specification of

**Comment by scarletsky** [comment]
@stephencookdev I did some tests:

## 1. No any changes:

```
 SMP  ⏱
General output time took 26.73 secs

 SMP  ⏱  Plugins
HtmlWebpackPlugin took 5.74 secs
CaseSensitivePathsPlugin took 3.082 secs
LodashModuleReplacementPlugin took 0.166 secs
ProgressPlugin took 0.079 secs
MiniCssExtractPlugin took 0.04 secs
ProvidePlugin took 0.002 secs
DefinePlugin took 0 secs
```

## 2. Changes the `optimization` in `webpack.config.js`:

```js
let optimization = {
    minimize: true,
    minimizer: [
        new UglifyJsPlugin()
    ].map(plugin => env.MEASURE ? new WrappedPlugin(plugin, plugin.constructor.name, smp) : plugin)
};
```

```
 SMP  ⏱
General output time took 26.73 secs

 SMP  ⏱  Plugins
HtmlWebpackPlugin took 5.74 secs
CaseSensitivePathsPlugin took 2.99 secs
LodashModuleReplacementPlugin took 0.202 secs
ProgressPlugin took 0.08 secs
MiniCssExtractPlugin took 0.05 secs
ProvidePlugin took 0.002 secs
DefinePlugin took 0.002 secs
```


## 3. Changes the `wrap` function like this:

```js
 config.plugins = (config.plugins || []).concat(config.optimization.minimizer).map(plugin => {
  const pluginName =
    Object.keys(this.options.pluginNames || {}).find(
      pluginName => plugin === this.options.pluginNames[pluginName]
    ) ||
    (plugin.constructor && plugin.constructor.name) ||
    "(unable to deduce plugin name)";
  return new WrappedPlugin(plugin, pluginName, this);
});
```

```
 SMP  ⏱
General output time took 42.35 secs

 SMP  ⏱  Plugins
UglifyJsPlugin took 19.62 secs
HtmlWebpackPlugin took 5.73 secs
CaseSensitivePathsPlugin took 3.08 secs
LodashModuleReplacementPlugin took 0.145 secs
ProgressPlugin took 0.069 secs
MiniCssExtractPlugin took 0.054 secs
ProvidePlugin took 0.001 secs
DefinePlugin took 0 secs
```

## 4. Changes the `wrap` function like this:

```js
 config.plugins = (config.plugins || []).map(plugin => {
  const pluginName =
    Object.keys(this.options.pluginNames || {}).find(
      pluginName => plugin === this.options.pluginNames[pluginName]
    ) ||
    (plugin.constructor && plugin.constructor.name) ||
    "(unable to deduce plugin name)";
  return new WrappedPlugin(plugin, pluginName, this);
});

if (config.optimization && config.optimization.minimize) {
  config.optimization.minimizer = config.optimization.minimizer.map(plugin => {
      return new WrappedPlugin(plugin, plugin.constructor.name, this);
  }) ;
}
```

```
 SMP  ⏱
General output time took 26.58 secs

 SMP  ⏱  Plugins
UglifyJsPlugin took 19.11 secs
HtmlWebpackPlugin took 5.53 secs
CaseSensitivePathsPlugin took 3.036 secs
LodashModuleReplacementPlugin took 0.149 secs
ProgressPlugin took 0.086 secs
MiniCssExtractPlugin took 0.055 secs
ProvidePlugin took 0.001 secs
DefinePlugin took 0.001 secs
```

**Comment by scarletsky** [comment]
I am confuse with test2 and test3.

I think test2 should work, but it does not output `UglifyJsPlugin`'s time.

And it seems test3 run twice `UglifyJsPlugin`, but I am not sure about it.

**Comment by Kruemelkatze** [comment]
I just tested your ***test4*** in one of my projects and it worked.

You should however not check for ```config.optimization.minimize```. People may not have set ```minimize=true``` in their production webpack config, as this is [true by default in production](https://webpack.js.org/configuration/optimization/#optimization-minimize), but this is set ***after*** the wrap method.

I replaced the check with```config.optimization.minimizer``` and it worked just fine for both cases with and without this setting.

**Comment by stephencookdev** [comment]
Thanks for your help with this @scarletsky and @Kruemelkatze :slightly_smiling_face: I agree that _**test4**_ but with the `if` statement changed to `if (config.optimization && config.optimization.minimizer) {` is probably the best way forwards here

Would one of you like to raise a PR to fix this?

**Comment by stephencookdev** [comment]
This is now merged in v1.3.0 :slightly_smiling_face:
